packages/firebase_auth/firebase_auth_web/lib/src/utils/web_utils.dart

```diff
@@ -242,6 +251,17 @@ auth_interop.OAuthCredential? convertPlatformCredential(
   }
 
   if (credential is OAuthCredential) {
+    if(credential.rawNonce != null) {
```

Comment:
Once you've created `OAuthCredentialOptions`, initialize here, remove this [code](https://github.com/FirebaseExtended/flutterfire/pull/6597/files#diff-a522fde29aeb319e6149a657c49c90dcce5c910e0e1166a1b37a0f00b4360885R254-R264) and update the call method to something like:
```dart
return auth_interop.OAuthProvider(credential.providerId).credential(
       OAuthCredentialOptions,
      );
```
as documented in Firebase JS SDK [here](https://firebase.google.com/docs/reference/js/firebase.auth.OAuthProvider#credential)

Reply:
If I change this code  I think I should also change this one and add an "options" field in addition of the existing. https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth_platform_interface/lib/src/providers/oauth.dart#L75
Othwerise I have to replace the existing fields with this new Option class. But if I do that, there will be a logic branch that will not be available.
The main difficulty is that we haven't Union Types in Dart. You can see my comment for your last change request of this PR. So what should I do? :)